net-utils/src/ip_echo_client.rs

```diff
@@ -112,30 +112,52 @@ pub(crate) const DEFAULT_RETRY_COUNT: usize = 5;
 /// `ip_echo_server_addr`. Tests must complete within timeout provided.
 /// Tests will run concurrently to avoid head-of-line blocking. Will return false on any error.
 /// This function may panic.
+/// All listening sockets should be bound to the same IP.
 pub(crate) async fn verify_all_reachable_tcp(
     ip_echo_server_addr: SocketAddr,
-    listeners: Vec<(u16, TcpListener)>,
+    listeners: Vec<TcpListener>,
     timeout: Duration,
 ) -> bool {
     if listeners.is_empty() {
         warn!("No ports provided for verify_all_reachable_tcp to check");
         return true;
     }
+    let bind_address = listeners[0]
+        .local_addr()
+        .expect("Sockets should be bound")
+        .ip();
+
+    for listener in listeners.iter() {
+        let local_binding = listener.local_addr().expect("Sockets should be bound");
+        assert_eq!(
+            local_binding.ip(),
+            bind_address,
+            "All sockets should be bound to the same IP"
+        );
+    }
     let mut checkers = Vec::new();
     let mut ok = true;
 
     // Chunk the port range into slices small enough to fit into one packet
     for chunk in &listeners.into_iter().chunks(MAX_PORT_COUNT_PER_MESSAGE) {
-        let (ports, listeners): (Vec<_>, Vec<_>) = chunk.unzip();
+        let listeners = chunk.collect_vec();
+        let ports = listeners
+            .iter()
+            .map(|l| l.local_addr().unwrap().port())
```

Comment:
```suggestion
            .map(|l| l.local_addr().expect("Sockets should be bound").port())
```
I know that you've checked before but I feel confused when see unwrap

Reply:
Well in this case unwrap is "safe" but I guess expect does not hurt either.